var/spack/repos/builtin/packages/foam-extend/package.py

```diff
@@ -0,0 +1,265 @@
+from spack import *
+from spack.environment import *
+
+import multiprocessing
+import subprocess
+import re
+import os
+
+
+class FoamExtend(Package):
+    """The foam-extend project is a fork of the OpenFOAM open source library
+      for Computational Fluid Dynamics (CFD)."""
+
+    version('3.2', git='http://git.code.sf.net/p/foam-extend/foam-extend-3.2')
+    version('3.1', git='http://git.code.sf.net/p/foam-extend/foam-extend-3.1')
+    version('3.0', git='http://git.code.sf.net/p/foam-extend/foam-extend-3.0')
+
+    variant('paraview', default=False, description='Enable ParaFOAM')
+    variant('scotch', default=True, description='Activate Scotch as a possible decomposition library')
+    variant('ptscotch', default=True, description='Activate PT-Scotch as a possible decomposition library')
+    variant('metis', default=True, description='Activate Metis as a possible decomposition library')
+    variant('parmetis', default=True, description='Activate Parmetis as a possible decomposition library')
+    variant('parmgridgen', default=True, description='Activate Parmgridgen support')
+
+    supported_compilers = {'clang': 'Clang', 'gcc': 'Gcc', 'intel': 'Icc'}
+
+    depends_on('mpi')
+    depends_on('cmake')
+    depends_on('python')
+    depends_on('flex@:2.5.99')
+    depends_on('bison')
+    depends_on('m4')
+    depends_on('hwloc')
+    depends_on('zlib')
+
+    depends_on('scotch ~ metis', when='~ptscotch+scotch')
+    depends_on('scotch ~ metis + mpi', when='+ptscotch')
+    depends_on('metis@5:', when='+metis')
+    depends_on('parmetis', when='+parmetis')
+    depends_on('parmgridgen', when='+parmgridgen')
+
+    depends_on('paraview', when='+paraview')
+    depends_on('qt@:4', when='+paraview')
```

Comment:
`cmake` is also used by the tests, but they are not compiled now so it can be removes,
`zlib` is explicitly used in the compile options, I don't know if it could be patched to be removed or if it is a real dependency.
For `bison` they are wmake rules defined using it, but they do not seam to be used
`m4` seams to be used in some tutorials so not mandatory to build *foam*
For the other one as you mentioned it seams to be only to compile third-parties.
I try to compile with the strict minimum. I just defined what was mentioned in prefs.sh.example that was a bit to much I guess.